examples/chip-tool/commands/pairing/RequestCommissioningCommand.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "RequestCommissioningCommand.h"
+
+#include <app/server/Server.h>
+#include <support/CodeUtils.h>
+
+using namespace ::chip;
+
+constexpr uint16_t kWaitDurationInSeconds = 30;
+
+CHIP_ERROR RequestCommissioningCommand::Run(NodeId localId, NodeId remoteId)
```

Comment:
I'm surprised that localId and remoteId are the right arguments for this. The UDC command sends the instance name (which is not the nodeId) to the IP and port specified in the SRV record of the selected commissioner. Is there code to derive those values from the remoteId in the "SendUserDirectedCommissioningRequest" command?

Reply:
> I'm surprised that localId and remoteId are the right arguments for this.

Those are the arguments that currently all commands get; the virtual `Run` method has those arguments.

Reply:
I did not really have a use for the passed down localId or remoteId, but didn't have a choice to stop them from being passed.
As for deriving the instance name, looking at https://github.com/project-chip/connectedhomeip/pull/6790/files I believe the GetCommissionableInstanceName() function could be used?